gcp_variant_transforms/libs/annotation/vep/vep_runner.py

```diff
@@ -0,0 +1,464 @@
+          self._create_actions(input_file, output_file))
+    api_request[_API_PIPELINE][_API_ACTIONS].append(
+        self._make_action('gsutil', '-q', 'cp',
+                          '/google/logs/output',
+                          output_log_path,
+                          flags=['ALWAYS_RUN']))
+    # pylint: disable=no-member
+    request = self._pipeline_service.pipelines().run(body=api_request)
+    operation_name = request.execute()['name']
+    return operation_name
+
+  def _check_and_write_to_output_dir(self, output_dir):
+    # type: (str) -> None
+    real_dir = self._format_dir_path(output_dir)
+    # NOTE(bashir2): We cannot use exists() because for example on GCS, the
+    # directory names are only symbolic and are not physical files.
+    match_results = filesystems.FileSystems.match(['{}*'.format(real_dir)])
+    if match_results and match_results[0].metadata_list:
+      raise ValueError('Output directory {} already exists.'.format(real_dir))
+    log_file = filesystems.FileSystems.create(
+        filesystems.FileSystems.join(output_dir, _GLOBAL_LOG_FILE))
+    # TODO(bashir2): Instead of just creating an empty file, log some
+    # information about how the VEP pipelines are executed.
+    log_file.close()
+
+  def _format_dir_path(self, dir_path):
+    # type: (str) -> str
+    """Returns `dir_name` possibly with an added '/' if not already included."""
+    return filesystems.FileSystems.join(dir_path, '')
+
+  def _get_output_log_path(self, output_dir, vm_index):
+    # type: (str, int) -> str
+    return '{}/logs/output_VM_{}'.format(output_dir, vm_index)
+
+  def _create_actions(self, input_file, output_file):
+    # type: (str, str) -> List
+    local_input_file = '/mnt/vep/{}'.format(_get_base_name(input_file))
+    return [
+        self._make_action('gsutil', '-q', 'cp', input_file, local_input_file),
+        self._make_action('rm', '-r', '-f', _LOCAL_OUTPUT_DIR),
+        self._make_action(_VEP_RUN_SCRIPT, local_input_file,
+                          _LOCAL_OUTPUT_FILE),
+        # TODO(bashir2): When the output files are local, the output directory
+        # structure should be created as well otherwise gsutil fails.
+        self._make_action('gsutil', '-q', 'cp', _LOCAL_OUTPUT_FILE,
+                          output_file)]
+
+
+class _SingleWorkerActions(object):
+  """Holds information about actions on a single virtual machine.
+
+  This is a pure data object and atributes can be accessed directly.
+  """
+
+  def __init__(self):
+    self.disk_size_bytes = 0
+    # `io_map` is a map from an input file to its corresponding output file.
+    self.io_map = {}  # type: Dict[str, str]
+
+  def __repr__(self):
+    return 'disk_size_bytes= {} , io_map= {}'.format(
+        self.disk_size_bytes, str(self.io_map))
+
+
+# TODO decide whether we want to unit-test this class in isolation and if so
+# move this and _SingleWorkerActions to a separate module.
+class _PipelinesExecutionInfo(object):
```

Comment:
My main concern is that making this a class implies that it's doing some 'work' and actually has some state, whereas the 'state' can just be achieved as a return value. I don't feel strongly about it though, so feel free to leave as is.